File: 0347-top-k-frequent-elements/0347-top-k-frequent-elements.py
class Solution(object):
    def topKFrequent(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: List[int]
        """
        count = Counter(nums)
        arr = []
        for key, v in count.items():
            arr.append((-1*v,key))
        heapq.heapify(arr)
        ans = []
        while k:
            value, val = heapq.heappop(arr)
            ans.append(val)
            print("hi")
            k-=1
        return ans 

       

        
        

File: 0347-top-k-frequent-elements/0347-top-k-frequent-elements.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        unordered_map<int, int> freq;

        for (int num : nums) {
            freq[num]++;
        }

        vector<pair<int, int>> hold;

        for (auto &p : freq) {
            hold.push_back({p.second, p.first}); 
        }

        sort(hold.rbegin(), hold.rend());

        vector<int> ans;

        for (int i = 0; i < k; i++) {
            ans.push_back(hold[i].second);
        }

        return ans;
    }
};